[zh-cn]: translate for setRequestHeader (#21739)
mdn/translated-content: files/zh-cn/web/api/xmlhttprequest/setrequestheader/index.md

---
title: XMLHttpRequest：setRequestHeader() 方法
slug: Web/API/XMLHttpRequest/setRequestHeader
l10n:
  sourceCommit: 9c78a44b9321fcd3fbe63d6f5b61ed749c2fa261
---

{{APIRef("XMLHttpRequest API")}} {{AvailableInWorkers("window_and_worker_except_service")}}

> {{domxref("XMLHttpRequest")}} 的方法 **`setRequestHeader()`** 设置 HTTP 请求头部的值。此方法必须在 {{domxref("XMLHttpRequest.open", "open()")}} 方法和 {{domxref("XMLHttpRequest.send", "send()")}} 之间调用。如果多次对同一个请求头赋值，只会生成一个合并了多个值的请求头。

Reviewer: ```suggestion
{{domxref("XMLHttpRequest")}} 的 **`setRequestHeader()`** 方法设置 HTTP 请求头部的值。此方法必须在 {{domxref("XMLHttpRequest.open", "open()")}} 方法和 {{domxref("XMLHttpRequest.send", "send()")}} 之间调用。如果多次对同一个请求头赋值，只会生成一个合并了多个值的请求头。
```

在第一次调用 `setRequestHeader()` 之后的每次调用，指定文本会附加到存在的标头内容的末尾。

> 如果没有设置 {{HTTPHeader("Accept")}} 属性，则此发送出{{domxref("XMLHttpRequest.send", "send()")}} 的值为此属性的默认值：`*/*`。

Reviewer: ```suggestion
如果没有设置 {{HTTPHeader("Accept")}} 标头，则在调用 {{domxref("XMLHttpRequest.send", "send()")}} 时，会随请求发送一个类型为 `"*/*"` 的 `Accept` 标头。
```

> 如果没有设置 {{HTTPHeader("Accept")}} 属性，则此发送出{{domxref("XMLHttpRequest.send", "send()")}} 的值为此属性的默认值：`*/*`。

Reviewer: ```suggestion
如果没有设置 {{HTTPHeader("Accept")}} 标头，则在调用 {{domxref("XMLHttpRequest.send", "send()")}} 时，会随请求发送一个类型为 `"*/*"` 的 `Accept` 标头。
```

Author: 这个重复了。

> 安全起见，有些请求头的值只能由 user agent 设置：{{Glossary("Forbidden_header_name", "forbidden header names", 1)}}。任何从前端 JavaScript 代码尝试为其中一个标头设置值都将被忽略，不会出现警告或错误。

Reviewer: https://developer.mozilla.org/zh-CN/docs/Glossary/Forbidden_header_name

```suggestion
安全起见，有些{{Glossary("Forbidden_header_name", "禁止修改的标头")}}的值只能由用户代理控制。任何通过前端 JavaScript 代码为其中一个标头设置值的尝试都将被忽略，不会出现警告或错误。
```

另外，[`Authorization`](/zh-CN/docs/Web/HTTP/Headers/Authorization) HTTP 标头可以添加到请求中，但如果请求被跨源重定向，则会被删除。

> > **备注：** 自定义一些 header 属性进行跨域请求时，可能会遇到"**not allowed by Access-Control-Allow-Headers in preflight response**"，你可能需要在你的服务端设置 "Access-Control-Allow-Headers"。

Reviewer: ```suggestion
> **备注：** 对于自定义字段，在跨域请求时可能遇到“**not allowed by Access-Control-Allow-Headers in preflight response**”异常，你可能需要在你的服务端设置 {{HTTPHeader("Access-Control-Allow-Headers")}}。
```

## 语法

```js-nolint
setRequestHeader(header, value)
```

### 参数

- `header`
  - : 要设置的标头的名称。
- `value`
  - : 要设置的标头正文的值。

### 返回值

> 无 ({{jsxref("undefined")}})。

Reviewer: ```suggestion
无（{{jsxref("undefined")}}）。
```
